utils/path.go

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package utils
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// SanitizePath will remove any relative directory changes and return the path
+// with them removed. Trailing separators are also removed.
+func SanitizePath(path string) string {
+	splitPath := strings.Split(path, string(os.PathSeparator))
+	fmt.Println(splitPath)
+	safePath := ""
+	for _, p := range splitPath {
+		if p != ".." && p != "~" {
+			safePath = filepath.Join(safePath, p)
+		}
+	}
+	return safePath
+}
```

Comment:
I think it's a little weird that "sanitizing" the path might actually return a path with a completely different meaning. Maybe it would be better to use something like this:

```go
func PathTraversesUpward(path string) bool {
    return strings.HasPrefix(filepath.Clean(path), "..")
}
```

Then error out if the tarball has a path that traverses upward. That would allow paths like "a/../b" to have the correct behavior too.

Reply:
doing something like we have for SAML doesn't work for this?
https://github.com/mattermost/platform/blob/master/app/saml.go#L35

Reply:
@ccbrown: Yeah, I guess it makes more sense to fail in this case rather than try to parse the "correct" path
@enahum: I think what you linked only worries about the filename when we're more worried about the path here